### PR TITLE
[fix] Fixing time controls when switching datasource types

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

This PR updates the temporal controls when switching datasource types in explorer which were missing as the Druid connector and SQLAlchemy connector have different fields. When we set the datasource upon change we also need to update the controls (indirectly via the form-data) leveraging the state changes.

Thanks @graceguo-supercat for all the help, I would have been completely lost without you. 

#### Before

Screen Shot 2019-12-03 at 6 49 18 PM

#### After 

Screen Shot 2019-12-03 at 6 58 01 PM

### TEST PLAN

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @betodealmeida @etr2460 @graceguo-supercat @michellethomas

### CODE REVIEW

File: superset/assets/src/explore/reducers/exploreReducer.js [hunk 1]
```diff
@@ -36,9 +36,31 @@ export default function exploreReducer(state = {}, action) {
       };
     },
     [actions.SET_DATASOURCE]() {
+      const newFormData = {...state.form_data};
+      if (action.datasource.type != state.datasource.type) {
```
Comment (etr2460) [inline superset/assets/src/explore/reducers/exploreReducer.js]: nit: `!==`

File: superset/assets/src/explore/reducers/exploreReducer.js [hunk 2]
```diff
@@ -36,9 +36,31 @@ export default function exploreReducer(state = {}, action) {
...
+        if (action.datasource.type == "table") {
```
Comment (etr2460) [inline superset/assets/src/explore/reducers/exploreReducer.js]: nit: `===`

File: superset/assets/src/explore/reducers/exploreReducer.js (hunk 1 above)
Comment (etr2460) [inline superset/assets/src/explore/reducers/exploreReducer.js]: a comment explaining that this maps equivalent params between sqla and druid might be worthwhile

File: superset/assets/src/explore/reducers/exploreReducer.js [hunk 3]
```diff
@@ -36,9 +36,31 @@ export default function exploreReducer(state = {}, action) {
...
+        }
+      }
+      const newState = {
+        ...state,
+        datasource: action.datasource,
+        datasource_id: action.datasource.id,
+        datasource_type: action.datasource.type,
+      };
       return {
         ...state,
```
Comment (etr2460) [inline superset/assets/src/explore/reducers/exploreReducer.js]: shouldn't this be `...newState` now?

**Review by etr2460 — CHANGES_REQUESTED** [review]
looks like you need to fix some linting errors. you can see them by running `npm run lint` and auto fix some of them with `npm run lint-fix`

File: superset/assets/src/explore/reducers/exploreReducer.js (hunk 3 above)
Comment (john-bodley) [reply superset/assets/src/explore/reducers/exploreReducer.js]: No. The `newState` is only required to update the controls.

File: superset/assets/src/explore/reducers/exploreReducer.js (hunk 1 above)
Comment (john-bodley) [reply superset/assets/src/explore/reducers/exploreReducer.js]: @etr2460 there's no mapping of params merely enabling/disabling.

File: superset/assets/src/explore/reducers/exploreReducer.js [hunk 4]
```diff
@@ -36,9 +36,31 @@ export default function exploreReducer(state = {}, action) {
...
+        form_data: newFormData,
         datasource: action.datasource,
```
Comment (etr2460) [inline superset/assets/src/explore/reducers/exploreReducer.js]: remove this line

**Review by etr2460 — APPROVED** [review]
lgtm, thanks!

**Pull request merged**